streams-verbs/src/main/java/org/apache/streams/verbs/VerbDefinitionResolver.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.streams.verbs;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import org.apache.streams.pojo.json.Activity;
+import org.apache.streams.pojo.json.ActivityObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+public class VerbDefinitionResolver {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(VerbDefinitionResolver.class);
+
+    protected Set<VerbDefinition> verbDefinitionSet;
+
+    public VerbDefinitionResolver() {
+        // get with reflection
+    }
+
+    public VerbDefinitionResolver(Set<VerbDefinition> verbDefinitionSet) {
+        this();
+        this.verbDefinitionSet = verbDefinitionSet;
+    }
+
+    public List<VerbDefinition> matchingVerbDefinitions(Activity activity) {
+
+        Set<VerbDefinition> candidates = Sets.newConcurrentHashSet(verbDefinitionSet);
```

Comment:
This does not clone the individual VerbDefinitions so when you mutate them below, it is changing those definitions for all users of this instance, or any other object that has a reference to the same VerbDefinitions, right?